cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/opencv-control.tsx

```diff
@@ -215,6 +241,69 @@ class OpenCVControlComponent extends React.PureComponent<Props & DispatchToProps
         return points;
     }
 
+    private runImageModifier():void {
+        const { activeImageModifiers } = this.state;
+        if (activeImageModifiers.length !== 0) {
+            const {
+                frameData, states, curZOrder, canvasInstance, frame,
+            } = this.props;
+            canvasInstance.configure({ forceFrameUpdate: true });
+            const canvas: HTMLCanvasElement | undefined = window.document.getElementById('cvat_canvas_background') as
+                | HTMLCanvasElement
+                | undefined;
+            if (!canvas) {
+                throw new Error('Element #cvat_canvas_background was not found');
+            }
+            const { width, height } = canvas;
+            const context = canvas.getContext('2d');
+            if (!context) {
+                throw new Error('Canvas context is empty');
+            }
+            const imageData = context.getImageData(0, 0, width, height);
+            const newImageData = activeImageModifiers.reduce((oldImageData, activeImageModifier) =>
+                activeImageModifier.modifier.processImage(oldImageData, frame), imageData);
+            frameData.imageData = newImageData;
+            canvasInstance.setup(frameData, states, curZOrder);
+            canvasInstance.configure({ forceFrameUpdate: false });
```

Comment:
Probably better to have done in finally block

Reply:
Done